lib/chef/provisioning/fog_driver/driver.rb

```diff
@@ -403,7 +404,9 @@ def wait_until_ready(action_handler, machine_spec, machine_options, server)
       if !server.ready?
         if action_handler.should_perform_actions
           action_handler.report_progress "waiting for #{machine_spec.name} (#{server.id} on #{driver_url}) to be ready ..."
-          server.wait_for(remaining_wait_time(machine_spec, machine_options)) { ready? }
+          retryable_request do
+            server.wait_for(remaining_wait_time(machine_spec, machine_options)) { ready? }
```

Comment:
Because it can potentially take a long time for this to complete now, I like to add `action_handler.report_progress "Waited #{current}/#{total}s for server to become ready" so users get a status update as the retryable process loops